File: 01_Singleton/TS/singleton.ts
class Singleton {
    static uniqueInstance: Singleton;
  
    constructor() {
      if (Singleton.uniqueInstance !== undefined) {
        return Singleton.uniqueInstance
      }
      Singleton.uniqueInstance = this;
    }
  }
  
  var instanceOne = new Singleton()
  var instanceTwo = new Singleton()
  
  console.log(instanceOne === instanceTwo)

File: 01_Singleton/CPP/singleton.cpp
#include "singleton.h"
Singleton* Singleton::uniqueInstance = nullptr; 

Singleton::Singleton(){
    
}

Singleton* Singleton::Instance () {
    if (uniqueInstance == nullptr) {      // Check ob bereits konstruiert
        uniqueInstance = new Singleton;     // Konstruktion
    }
    return uniqueInstance;                // Rückgabe des Objekts
} 